pkg/report/sarif.go

```diff
@@ -341,10 +341,20 @@ func ToPathUri(input string, resultClass types.ResultClass) string {
 	return strings.ReplaceAll(strings.ReplaceAll(input, "\\", "/"), "git::https:/", "")
 }
 
-func (sw *SarifWriter) getLocations(name, version, path string, pkgs []ftypes.Package) []location {
+func (sw *SarifWriter) getLocations(name, version, path string, pkgs []ftypes.Package, vulnLocations []ftypes.Location) []location {
```

Comment:
We don't need location cache and package check now.

What do you thin about this update for this function:

```go
func getLocations(vulnLocations []ftypes.Location) []location {
	if len(vulnLocations) == 0 {
		return nil
	}

	var locs []location
	for _, loc := range vulnLocations {
		locs = append(locs, location{
			startLine: loc.StartLine,
			endLine:   loc.EndLine,
		})
	}
	return locs
}
```

Reply:
@DmitriyLewen Initially, I considered this approach. However, after reviewing the [test cases](https://github.com/aquasecurity/trivy/blob/main/pkg/report/sarif_test.go#L25-L184), which uses `dbTypes.Vulnerabilities` instead of `types.DetectedVulnerabilities`, I decided that implementing backward compatible changes would be beneficial. If you're confident that this won't impact the approach, I'll proceed with updating it to match the example you provided.

Reply:
If another app generates `Report` using Trivy packages - these changes will have no effect.
But if this app creates `Report` on its own, it will need to simply copy `Location` from `Package` to `Vulnerabilities Detected`.

So I think these changes won't be a problem.